src/postcss.ts

```diff
@@ -0,0 +1,206 @@
+import postcss, { Rule, AtRule, Root, Node, Declaration } from 'postcss';
+import modules from 'postcss-modules';
+import selectorParser, { ClassName } from 'postcss-selector-parser';
+import valueParser from 'postcss-value-parser';
+
+/** The pair of line number and column number. */
+export type Position = {
+  /** The line number in the source file. It is 1-based. */
+  line: number;
+  /** The column number in the source file. It is 1-based. */
+  column: number;
+};
+
+/** The location of class selector. */
+export type Location = {
+  filePath: string;
+  /** The inclusive starting position of the node's source. */
+  start: Position;
+  /** The inclusive ending position of the node's source. */
+  end: Position;
+};
+
+/**
+ * The name of exported token by the source file excluding the imported from other sheets.
+ * @param ast The root node of the source file.
+ * @returns The exported token names.
+ */
+export async function generateLocalTokenNames(ast: Root): Promise<string[]> {
+  return new Promise((resolve, reject) => {
+    postcss()
+      .use(
+        modules({
+          getJSON: (_cssFileName, json) => {
+            resolve(Object.keys(json));
+          },
+        }),
+      )
+      // NOTE: `process` modifies ast, so clone it.
+      .process(ast.clone())
```

Comment:
When run tests, postcss-modules outputs a warning. Probably because it is trying to track the file specified in the composes property. The phase in which `generateLocalTokenNames` is executed should not track composes. I will fix this later.

```console
$ pnpm test
 PASS  test/postcss.spec.ts
  ● Console

    console.log
      [Error: ENOENT: no such file or directory, open '/test/other.css'] {
        errno: -2,
        code: 'ENOENT',
        syscall: 'open',
        path: '/test/other.css'
      }

      at node_modules/.pnpm/postcss-modules@4.3.1_postcss@8.4.14/node_modules/postcss-modules/build/css-loader-core/parser.js:77:23
          at async Promise.all (index 0)
```